Dockerfile

```
@@ -3,31 +3,32 @@ FROM rust:latest as builder
 RUN USER=root cargo new --bin lychee
 WORKDIR /lychee
 
-# Just copy the Cargo.toml files and trigger 
+# Just copy the Cargo.toml files and trigger
 # a build so that we compile our dependencies only.
 # This way we avoid layer cache invalidation
 # if our dependencies haven't changed,
 # resulting in faster builds.
 COPY lychee-bin/Cargo.toml lychee-bin/Cargo.toml
 COPY lychee-lib/Cargo.toml lychee-lib/Cargo.toml
-RUN cargo build --release
-RUN rm src/*.rs
+RUN cargo build --release \
+    && rm src/*.rs
 
 # Copy the source code and run the build again.
 # This should only compile lychee itself as the
 # dependencies were already built above.
-ADD . ./
-RUN rm ./target/release/deps/lychee*
-RUN cargo build --release
+COPY . ./
+RUN rm ./target/release/deps/lychee* \
+    && cargo build --release
 
-
-# Our production image starts here, which uses 
+# Our production image starts here, which uses
 # the files from the builder image above.
 FROM debian:buster-slim
 
 RUN apt-get update \
-    && apt-get install -y ca-certificates tzdata \
-    && rm -rf /var/lib/apt/lists/*
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    --no-install-recommends ca-certificates tzdata \
+    && rm -rf /var/lib/apt/lists/* \
+    /var/cache/debconf/*
```

```suggestion
    /var/cache/debconf/* \
    /var/cache/apt/*
```
Contains downloaded DEB packages and the two ~30 MiB binary and source package cache files.

---

Did you try to build the container with this modification? Can you show me where those ~30MB files are?
Finally, maybe `apt-get clean` would be more appropriate but I have no strong opinion on which is better here.

---

Ah, the cache seems to be disabled on Debian's Docker image and automatically cleaned after APT update and package installs (`dpkg` invocation): https://github.com/debuerreotype/debuerreotype/blob/c162fca/scripts/debuerreotype-minimizing-config#L88-L108
We could verify whether this config file `/etc/apt/apt.conf.d/docker-clean` exists, to be sure. `apt-get clean` does exactly the same but is then not required either.

Check out `/var/cache/apt/pkgcache.bin` and `/var/cache/apt/scrpkgcache.bin` on every other Debian system.

Also interesting the other config:
> 		# Since Docker users using "RUN apt-get update && apt-get install -y ..." in
>		# their Dockerfiles don't go delete the lists files afterwards...

We do 😄.

---

Interesting. I would suggest we keep those cleaning mechanism just in case we change the base image in the future. As you points out, a "normal" Debian installation would require those cleaning. Plus, the cost of these actions is low.

There is also a way of preventing to push an unnecessary big image in the registry by using a [dive](https://github.com/wagoodman/dive) check step in the CI.